linux/devdoc/threadpool_linux_requirements.md

```diff
@@ -400,11 +408,17 @@ MOCKABLE_FUNCTION(, THREADPOOL_WORK_ITEM_HANDLE, threadpool_create_work_item, TH
 
 **S_R_S_THREADPOOL_LINUX_05_002: [** If `work_function` is `NULL`, `threadpool_create_work_item` shall fail and return a `NULL` value. **]**
 
-**S_R_S_THREADPOOL_LINUX_05_003: [** `threadpool_create_work_item` shall allocate memory for `threadpool_work_item` of type `THREADPOOL_WORK_ITEM_HANDLE`. **]**
+**S_R_S_THREADPOOL_LINUX_05_003: [** `threadpool_create_work_item` shall call `sm_exec_begin`. **]**
+
+**S_R_S_THREADPOOL_LINUX_05_004: [** If `sm_exec_begin` returns `SM_EXEC_REFUSED`, `threadpool_create_work_item` shall fail and return a non-zero value. **]**
+
+**S_R_S_THREADPOOL_LINUX_05_005: [** `threadpool_create_work_item` shall allocate memory for `threadpool_work_item` of type `THREADPOOL_WORK_ITEM_HANDLE`. **]**
+
+**S_R_S_THREADPOOL_LINUX_05_006: [** If during the initialization of `threadpool_work_item`, `malloc` fails then log the error and threadpool_create_work_item shall return `NULL`. **]**
```

Comment:
05_006 and 05_004 can be combined into one requirement that says 

"If there are any failures, `threadpool_create_work_item` shall fail and return a non-zero value"

This is how a lot of other modules do it (although not all) and seems to be the preferred way to spec the failure case.